recipes/glfw/all/conanfile.py

```diff
@@ -217,6 +219,17 @@ def package_info(self):
                 "CoreServices", "Foundation", "IOKit",
             ])
 
+        # Starting with version 3.4, glfw loads the platform libraries at runtime
+        self.cpp_info.requires = []
+        if Version(self.version) < "3.4":
+            if self.options.get_safe("vulkan_static"):
+                self.cpp_info.requires.extend(["vulkan-loader::vulkan-loader"])
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                if self.options.get_safe("with_x11", True):
+                    self.cpp_info.requires.extend(["xorg::x11"])
+            if self.options.get_safe("with_wayland"):
+                self.cpp_info.requires.extend(["wayland::wayland", "xkbcommon::xkbcommon"])
```

Comment:
```suggestion
        if self.options.get_safe("vulkan_static"):
            self.cpp_info.requires.extend(["vulkan-loader::vulkan-loader"])
        if self.settings.os in ["Linux", "FreeBSD"]:
            if self.options.get_safe("with_x11", True):
                self.cpp_info.requires.extend(["xorg::x11"])
        if self.options.get_safe("with_wayland"):
            self.cpp_info.requires.extend(["wayland::wayland", "xkbcommon::xkbcommon"])
```
I don't think the version check is necessary as it's already handled by the deleted options in `configure()`, no?

Reply:
I think the version check is needed because even when you specify `with_x11=True` or `with_wayland=True`, the glfw library does not require linking with these after version 3.4, which implemented the runtime platform detection.